Ical.Net/Evaluation/RecurringEvaluator.cs

```diff
@@ -40,42 +41,38 @@ public RecurringEvaluator(IRecurrable obj)
     /// <param name="periodStart">The beginning date of the range to evaluate.</param>
     /// <param name="periodEnd">The end date of the range to evaluate.</param>
     /// <param name="includeReferenceDateInResults"></param>
-    protected HashSet<Period> EvaluateRRule(IDateTime referenceDate, DateTime periodStart, DateTime periodEnd, bool includeReferenceDateInResults)
+    protected IEnumerable<Period> EvaluateRRule(IDateTime referenceDate, DateTime? periodStart, DateTime? periodEnd, bool includeReferenceDateInResults)
     {
         if (Recurrable.RecurrenceRules == null || !Recurrable.RecurrenceRules.Any())
-        {
-            return new HashSet<Period>();
-        }
+            return [];
 
-        var periodsQuery = Recurrable.RecurrenceRules.SelectMany(rule =>
+        var periodsQueries = Recurrable.RecurrenceRules.Select(rule =>
         {
             var ruleEvaluator = rule.GetService(typeof(IEvaluator)) as IEvaluator;
             if (ruleEvaluator == null)
             {
                 return Enumerable.Empty<Period>();
             }
             return ruleEvaluator.Evaluate(referenceDate, periodStart, periodEnd, includeReferenceDateInResults);
-        });
+        }).ToList();
```

Comment:
no deferred execution in order to convert the result to a list to ensure that all recurrence rules are fully evaluated before any further processing?